pydm/widgets/rules_editor.py

```diff
@@ -0,0 +1,479 @@
+import os
+import json
+import webbrowser
+
+from ..PyQt import QtGui, QtCore, QtDesigner
+from ..utilities.iconfont import IconFont
+
+
+class RulesEditor(QtGui.QDialog):
+    """
+    QDialog for user-friendly editing of the Rules in a widget inside the Qt
+    Designer.
+
+    Parameters
+    ----------
+    widget : PyDMWidget
+        The widget which we want to edit the `rules` property.
+    """
+
+    def __init__(self, widget, parent=None):
+        super(RulesEditor, self).__init__(parent)
+
+        self.widget = widget
+        self.lst_rule_item = None
+        self.loading_data = True
+
+        self.available_properties = widget.RULE_PROPERTIES
+        self.default_property = widget.DEFAULT_RULE_PROPERTY
+
+        self.setup_ui()
+
+        try:
+            self.rules = json.loads(widget.rules)
+        except:
+            self.rules = []
+
+        for ac in self.rules:
+            self.lst_rules.addItem(ac.get("name", ''))
+
+    def setup_ui(self):
```

Comment:
In general I like to do as much of the static UI creation in Desginer then use `pydm.Display`. Keeps the business logic separated from the UI. I know this is a matter of taste, but I do think it is a design paradigm we should be reinforcing.

Reply:
Got it! In the pydm code we tend to not use the `.ui` files with some small exceptions... that is why I decided to go with the code to setup the ui...